File: 1899-count-items-matching-a-rule/count-items-matching-a-rule.py
class Solution(object):
    def countMatches(self, items, ruleKey, ruleValue):
        ruleKeys = ['type', 'color', 'name']
        count = 0
        index = -1

        for i in range(len(ruleKeys)):
            if ruleKeys[i] == ruleKey:
                index = i

        for i in range(len(items)):
            if index != -1 and items[i][index] == ruleValue:
                count += 1
        return count

File: 1899-count-items-matching-a-rule/count-items-matching-a-rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function(items, ruleKey, ruleValue) {
    let rulekeyno;
    if(ruleKey === "type"){
        rulekeyno = 0;
    } else if(ruleKey === "color"){
        rulekeyno = 1;
    } else if(ruleKey === "name"){
        rulekeyno = 2;
    } else {
        return 0;
    }

    let matched = 0;
    for(let i=0; i<items.length; i++){
        if(items[i][rulekeyno] === ruleValue){
            matched += 1;
        }
    }

    return matched;
};